lib/elixir/lib/calendar/date.ex

```diff
@@ -119,6 +124,11 @@ defmodule Date do
     range(first, first_days, last, last_days, calendar, step)
   end
 
+  def range(%{calendar: calendar} = first, %Duration{} = duration) do
```

Comment:
shall we allow the keyword list version as well? :)

Reply:
i found that `range(~D[2020-01-01], [year: 1], 2)` looks a bit unergonomic, but definitely nothing speaking against allowing it 👍